src/containers/Folder.jsx

```diff
@@ -20,27 +21,28 @@ import DeleteConfirmation from '../containers/DeleteConfirmation'
 
 const isDir = attrs => attrs.type === 'directory'
 
-const TRASH_CONTEXT = 'trash'
-
 class Folder extends Component {
   componentWillMount () {
-    this.props.onMount()
+    if (!this.props.isFetching) {
+      this.props.onMount()
+    }
   }
 
   componentWillReceiveProps (newProps) {
-    if (this.props.params.file !== undefined && // we're not in the root dir
-      newProps.params.file !== this.props.params.file && // the route has changed
-      newProps.params.file !== newProps.folderId) { // but the folder has not been fetched
-      this.props.onRouteChange(newProps.params.file)
+    // if we're already fetching, that's because of a 'openFolder' call
+    // no need to check if the URL has changed
+    if (this.props.isFetching || newProps.isFetching) return
```

Comment:
We may return something more explicit because we don't use semicolons: `return null` or `return undefined`.